### DISCUSSION TITLE
Is it safe to use two databases of the same class with two different store files

### DISCUSSION THREAD

**Discussion by ycherniavskyi** [simolus3/drift#1110] [Q&A] [answered]
I am going to implement some type of restore from backup (which stores on the server) functionality.

I see the next steps to implement it:
1. On app start I check if `db_from_backup.sqlite` file exists
    * if exist, them rename `db.sqlite` file to `db_rollback.sqlite` (for possibility to rollback last backup restoration) and rename `db_from_backup.sqlite` to `db.sqlite`
2. Create `AppDatabase` with `db.sqlite` file to provide it via Provider.
3. On restore from backup action
    1. Download data
    2. Create `AppDatabase` with `db_from_backup.sqlite` file and use it locally
    3. Insert data in databse
    4. Close database
        * if any error on insert, then delete `db_from_backup.sqlite` file and report to the user
        * if no error, them restart app

During creating the second `AppDatabase` I got the next warning:
```
WARNING (moor): It looks like you've created the database classAppDatabase multiple times. When these two databases use the same QueryExecutor, race conditions will ocur and might corrupt the database. 
```

Do I understand that in the current case it is safe to create several `AppDatabase` instances and I just need to set `moorRuntimeOptions.dontWarnAboutMultipleDatabases = true`?

**Comment by simolus3** [comment, marked as answer]
Yes exactly, this falls under the "you know what you're doing" rule. The warning was added because some users accidentally opened their databases (under the same file) multiple times. This can break stream queries and cause issues with migrations, which is why the warning was added.
Using two instances of the same class with different underlying database files is not an issue at all.
